google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminSettings.java

```diff
@@ -89,14 +92,53 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
-  /** Create a new builder preconfigured to connect to the Bigtable emulator. */
+  /**
+   * Create a new builder preconfigured to connect to Bigtable emulator when hostname and port
+   * number are provided through BIGTABLE_EMULATOR_HOST environment variable.
+   */
+  public static Builder newBuilderForEmulator() {
+    String emulatorHostAndPort = System.getProperty(BIGTABLE_EMULATOR_HOST_ENV_VAR);
+    Preconditions.checkState(
+        emulatorHostAndPort != null,
+        "BIGTABLE_EMULATOR_HOST not found in environment variable. Please use "
+            + "newBuilderForEmulator(hostname, port)");
+
+    String[] hostPort = emulatorHostAndPort.split(":");
+    Preconditions.checkArgument(
+        hostPort.length == 2,
+        "Malformed "
+            + BIGTABLE_EMULATOR_HOST_ENV_VAR
+            + " environment variable: "
+            + emulatorHostAndPort
+            + ". Expecting host:port.");
```

Comment:
I think this will fail for ipv6 addresses. It would be better to do use last index of

Reply:
Thanks for the feedback, I have just updated this change. Please have a look.